graspologic/match/solver.py

```diff
@@ -0,0 +1,734 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import time
+import warnings
+from functools import wraps
+from typing import Callable, Optional
+
+import numpy as np
+from beartype import beartype
+from numba import njit
+from ot import sinkhorn
+from packaging import version
+from scipy import __version__ as scipy_version
+from scipy.optimize import linear_sum_assignment
+from scipy.sparse import csr_matrix
+from sklearn.utils import check_scalar
+
+if version.parse(scipy_version) >= version.parse("1.8.0"):
```

Comment:
You have this logic in the types already, you can do `from .types import csr_array` and keep the if/then in a single place

Reply:
✅ https://github.com/microsoft/graspologic/pull/960/commits/be27bd080ea13d7b31e865f70d255eed445d3a4d